[IntegTests] Fix Multi-AZ create/update integration test failures

EddyMM opened the pull request:
### Description of changes
* The MultiAZ create/update tests rely on creation of capacity reservations in different AZs to ensure instances are launched in different subnets/AZs. One of public AZs created by the `vpc_stack` fixture may be chosen randomly, making it unstable for use in these tests.

* These changes ensure the MultiAZ tests use AZs chosen from a consistent/deterministic list of subnets/AZs.


### Tests
* Ran test locally after reproducing the issue

```
test-suites:
  update:
    test_update.py::test_multi_az_create_and_update:
      dimensions:
      - oss:
        - alinux2
        regions:
        - eu-west-2
        schedulers:
        - slurm
```

```
{
    "all": {
        "total": 1,
        "skipped": 0,
        "failures": 0,
        "errors": 0,
        "succeeded": 1
    },
```

### References
* N/A

### Checklist
- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [ ] Check all commits' messages are clear, describing what and why vs how.
- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

NSsirena commented on tests/integration-tests/conftest.py:
```diff
@@ -1182,18 +1182,19 @@ def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_s
     public_subnets.remove(public_subnet)
-    availability_zone = boto3.resource("ec2").Subnet(public_subnet).availability_zone
-    availability_zone_2 = boto3.resource("ec2").Subnet(public_subnets[0]).availability_zone
+    default_or_random_public_az = boto3.resource("ec2").Subnet(public_subnet).availability_zone
```
I'd call_this `default_public_az` since it corresponds to what the test will receive when asking for the `default_public_az` . 
It is randomized when vpc_stack is created (at each test initialization) but it will remain constant for the entire test lifecycle.

NSsirena commented on tests/integration-tests/conftest.py:
```diff
@@ -1182,18 +1182,19 @@ def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_s
-    availability_zone = boto3.resource("ec2").Subnet(public_subnet).availability_zone
-    availability_zone_2 = boto3.resource("ec2").Subnet(public_subnets[0]).availability_zone
+    default_or_random_public_az = boto3.resource("ec2").Subnet(public_subnet).availability_zone
+    availability_zone_1 = boto3.resource("ec2").Subnet(public_subnets[0]).availability_zone
+    availability_zone_2 = boto3.resource("ec2").Subnet(public_subnets[1]).availability_zone
```
[Nitpick] This whole method and how these resources are created and used should likely be redesigned. But this is out-of-scope for this issue.

NSsirena commented on tests/integration-tests/conftest.py:
```diff
@@ -1182,18 +1182,19 @@ def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_s
     public_subnet = vpc_stack.get_public_subnet()
     public_subnets = vpc_stack.get_all_public_subnets().copy()
     public_subnets.remove(public_subnet)
```
We shouldn't do this.

The ODCR stack using the `default_az` is used by single_az tests;
while `public_subnets[0]` and `public_subnets[1]` are used in the MultiAZODCR that is used by multi_az tests.

So it is important that what we here consider `public_subnets[0]` is the same seen by the test while requesting `{{ public_subnets[0] }}`

EddyMM replied:
#4898 
Seems like it was a fix applied for a problem encountered making improvements in the test_runner. It appears to ensure that `public_subnets` contains non-default AZs (but the exact reason for doing this is unclear).

NSsirena replied:
Likely it could create failures in the code below where we initialize `multi_az_odcr_group` when the two values pointed to the same AZ

EddyMM committed:
[IntegTests] Fix Multi-AZ create/update integration test failures

The MultiAZ create/update tests rely on creation of capacity reservations in different AZs to ensure instances are launched in different subnets/AZs.
One of public AZs created by the vpc_stack fixture may be chosen randomly, making it unstable for use in these tests.

These changes ensure the MultiAZ tests use AZs chosed from a consistent/deterministic list of subnets/AZs.

Signed-off-by: Eddy Mwiti <eddmwiti@amazon.com>
